.storybook/lucid-docs-addon/PanelToggles.js

```diff
@@ -1,34 +1,39 @@
 import React from 'react';
 import Hoverable from './Hoverable';
+import addons, { makeDecorator } from '@storybook/addons';
+import ReactDOM from 'react-dom';
 
-export default class extends React.Component {
-	constructor(...args) {
-		super(...args);
-		this.onToggleLayout = this.onToggleLayout.bind(this);
-		this.onTogglePanel = this.onTogglePanel.bind(this);
-	}
-
-	onToggleLayout() {
-		const urlState = this.props.api.getUrlState();
-		this.props.api.setOptions({
-			addonPanelInRight: !urlState.panelRight,
-		});
-	}
+const portalElement = window.document.createElement('div');
+window.parent.document.body.appendChild(portalElement);
 
-	onTogglePanel() {
-		const urlState = this.props.api.getUrlState();
-		this.props.api.setOptions({
-			showAddonPanel: !urlState.addons,
-		});
-	}
+export const withPanelToggles = makeDecorator({
+	name: 'withPanelToggles',
+	parameterName: 'panelToggles',
+	skipIfNoParametersOrOptions: true,
+	wrapper: (storyFn, context) => {
+		const channel = addons.getChannel();
+		const panelPortal = ReactDOM.createPortal(
+			<PanelToggles
+				handleToggleLayout={() => {
+					channel.emit('lucid-docs-panel-layout-toggle');
+				}}
+				handleTogglePanel={() => {
+					channel.emit('lucid-docs-panel-hide-toggle');
+				}}
+			/>,
+			portalElement
+		);
+		return (
+			<div>
+				{storyFn(context)}
+				{panelPortal}
+			</div>
+		);
+	},
+});
 
+class PanelToggles extends React.Component {
 	render() {
-		const urlState = this.props.api.getUrlState();
-
-		if (urlState.full || !urlState.addons) {
```

Comment:
urlState.addons intended use is to control the visibility of the panel sidebar.  Using that here creates a bug where if the user hides the sidebar and later reloads the url we don't show the toggle component.  Example: https://docspot.adnxs.net/projects/lucid/latest/?selectedKind=DropMenu&selectedStory=stateless&full=0&addons=0&stories=1&panelRight=1&addonPanel=lucid-docs-panel-props

This code is effectively replaced by using the new panelToggle decorator.